src/responses.js

```diff
@@ -0,0 +1,179 @@
+const { fixArraySchema, hasFileRef, getRefFromInclude, searchDataStructure } = require('./util')
+const escapeJSONPointer = require('./escape_json_pointer')
+const isEqual = require('lodash.isequal')
+const http = require('http')
+
+const parseResponseSchema = (schema, openApi3) => {
+    if (!schema) return
+    if (openApi3 && hasFileRef(schema)){
+        return getRefFromInclude(schema) 
+    }
+    try {
+        const result = JSON.parse(schema);
+        delete result['$schema'];
+        fixArraySchema(result); // work around for Swagger UI / Editor
+        return result
+    } catch (e) { }
+}
+
+const parseResponseBody = (body, header, openApi3) => {
```

Comment:
Instead of explicitly passing in `openapi3`, let's use the `options` variable or `context` to derive the state